openmc/mesh.py

```diff
@@ -2749,30 +2729,155 @@ def write_data_to_vtk(
                         f"with dimensions {self.dimension}"
                     )
 
-            if volume_normalization:
+            with h5py.File(filename, "w") as f:
+
+                root = f.create_group("VTKHDF")
+                root.attrs["Version"] = (2, 1)
+                ascii_type = "UnstructuredGrid".encode("ascii")
+                root.attrs.create(
+                    "Type",
+                    ascii_type,
+                    dtype=h5py.string_dtype("ascii", len(ascii_type)),
+                )
+
+                # create hdf5 file structure
+                root.create_dataset(
+                    "NumberOfPoints", (0,), maxshape=(None,), dtype="i8"
+                )
+                root.create_dataset("Types", (0,), maxshape=(None,), dtype="uint8")
+                root.create_dataset("Points", (0, 3), maxshape=(None, 3), dtype="f")
+                root.create_dataset(
+                    "NumberOfConnectivityIds", (0,), maxshape=(None,), dtype="i8"
+                )
+                root.create_dataset("NumberOfCells", (0,), maxshape=(None,), dtype="i8")
+                root.create_dataset("Offsets", (0,), maxshape=(None,), dtype="i8")
+                root.create_dataset("Connectivity", (0,), maxshape=(None,), dtype="i8")
+
+                append_dataset(root["NumberOfPoints"], np.array([len(self.vertices)]))
+                append_dataset(root["Points"], self.vertices)
+                append_dataset(
+                    root["NumberOfConnectivityIds"],
+                    np.array([len(trimmed_connectivity)]),
+                )
+                append_dataset(root["Connectivity"], trimmed_connectivity)
+                append_dataset(root["NumberOfCells"], np.array([self.n_elements]))
+                append_dataset(root["Offsets"], offsets)
+
+                # VTK_TETRA type is known as DAGMC only supports tet meshes
+                append_dataset(
+                    root["Types"], np.full(self.n_elements, 10, dtype="uint8")
```

Comment:
Let's define `10` as variable as well. I think we may want to use a type defined by the `vtk` Python module, but I understand that writing this file version doesn't require the `vtk` module which may be nice so for now let's start with this.